### PR TITLE
[fmiweather] Discovery: advertising all discovered thing types

### PR DESCRIPTION
**Pull request by ssalonen**
As discussed in https://github.com/openhab/openhab-addons/issues/8971#issuecomment-723420761

Signed-off-by: Sami Salonen <ssalonen@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/discovery/FMIWeatherDiscoveryService.java [hunk 1]
```diff
@@ -58,7 +60,8 @@
 public class FMIWeatherDiscoveryService extends AbstractDiscoveryService {
     private final Logger logger = LoggerFactory.getLogger(FMIWeatherDiscoveryService.class);
 
-    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_OBSERVATION);
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/discovery/FMIWeatherDiscoveryService.java:63]: Or even shorter when using the Java 9 `Set.of(...)` methods ...

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/discovery/FMIWeatherDiscoveryService.java [hunk 2]
```diff
@@ -58,7 +60,8 @@
...
+            Arrays.asList(THING_TYPE_OBSERVATION, THING_TYPE_FORECAST));
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/discovery/FMIWeatherDiscoveryService.java:64]:
```suggestion
    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_OBSERVATION, THING_TYPE_FORECAST);
```

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
Thanks, nice change

**Review by cpmeister — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**